pkg/sippyserver/server.go

```diff
@@ -265,6 +269,132 @@ func (s *Server) detailed(w http.ResponseWriter, req *http.Request) {
 
 }
 
+type Tests map[string][]int
+
+func unpackTests(tests []testgridv1.Test) Tests {
+	result := make(Tests)
+	for _, t := range tests {
+		var statuses []int
+		for _, testResult := range t.Statuses {
+			for i := 0; i < testResult.Count; i++ {
+				statuses = append(statuses, testResult.Value)
+			}
+		}
+		result[t.Name] = statuses
+	}
+	return result
+}
+
+func hasOpenShiftTests(tests Tests, i int) bool {
+	for test, results := range tests {
+		if strings.HasPrefix(test, "openshift-tests.") {
+			if results[i] != 0 {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func testStatus(tests Tests, name string, i int) int {
+	results, ok := tests[name]
+	if !ok {
+		return 0
+	}
+	return results[i]
+}
+
+func (s *Server) jobs(w http.ResponseWriter, req *http.Request) {
+	reportName := req.URL.Query().Get("release")
+	jobFilterString := req.URL.Query().Get("jobFilter")
+
+	var jobFilter *regexp.Regexp
+	if len(jobFilterString) > 0 {
+		var err error
+		jobFilter, err = regexp.Compile(jobFilterString)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("jobFilter: %s", err), http.StatusBadRequest)
+			return
+		}
+	}
+
+	dashboardCoordinates, found := s.reportNameToDashboardCoordinates(reportName)
+	if !found {
+		releasehtml.WriteLandingPage(w, s.reportNames())
+		return
+	}
+
+	type jsonTest struct {
+		Name    string   `json:"name"`
+		Results []string `json:"results"`
+	}
+	type jsonJob struct {
+		Name        string     `json:"name"`
+		Timestamps  []int      `json:"timestamps"`
+		Tests       []jsonTest `json:"tests"`
+		BuildIDs    []string   `json:"build_ids"`
+		TestGridURL string     `json:"testgrid_url"`
+	}
+	type jsonResponse struct {
+		Jobs           []jsonJob `json:"jobs"`
+		LastUpdateTime time.Time `json:"last_update_time"`
+	}
+
+	testGridJobDetails, lastUpdateTime := testgridhelpers.LoadTestGridDataFromDisk(s.testReportGeneratorConfig.TestGridLoadingConfig.LocalData, dashboardCoordinates.TestGridDashboardNames, jobFilter)
+	response := jsonResponse{
+		LastUpdateTime: lastUpdateTime,
+	}
+	for _, job := range testGridJobDetails {
+		results := unpackTests(job.Tests)
+		overall, ok := results["Overall"]
+		if !ok {
+			continue
+		}
+		var statuses []string
+		for i, result := range overall {
+			var status string
+			switch result {
+			case 4:
+				status = "R" // running
+			case 1:
+				status = "S" // success
+				if !hasOpenShiftTests(results, i) {
+					status = "P" // passed
```

Comment:
any particular reason we care about this distinction?

Reply:
For (1) i guess i can appreciate that the "no-op" successes need to be ignored so they don't provide false indication that the job is generally passing when it reality it's mostly no-op'ing and then failing whenever it tries to do real work.  But if that's the case, once there is real work to do, you should start seeing a string of repeated failures, i think?

For (2) I would still expect them to be interesting to TRT because they still represent an indication about whether we are able to install/run the cluster on a platform we care about.  And more generally if there are jobs TRT does not care about I think i'd rather see TRT maintain an explicit list of "jobs we don't care about and why" and pass that in as some sort of jobexclusion filter to this endpoint.

Reply:
(1) Sometimes it has nothing to do, then it fails to check that it has nothing to do, and then it's green again as it has nothing to do. Example: https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/promote-release-openshift-machine-os-content-e2e-aws-4.8-ppc64le/1374136409908056064

That's what I've found using this distinction. This knowledge isn't super useful, so I can ditch it.

(2) I guess I need incorporate bits that identify infra problems and installer failures. That should give us a signal from these tests and we'll be able to see more detailed picture about failures:

* infra
* installer
* upgrade
* openshift-e2e
* other tests (osd, certs, arcconformance, etc.)

Reply:
![image](https://user-images.githubusercontent.com/443137/113287827-6340e080-92ee-11eb-8ab1-f5c4cd5290dd.png)

Updated, now it looks like this.